File: code/graphs/dfs-recursive.cpp
void dfs(Graph& g, int u, vector<bool>& seen) {
    seen[u] = true;                 // visit u
    for (int v : g.adj[u])          // each neighbor
        if (!seen[v])
            dfs(g, v, seen);        // recurse; backtrack on return
}

File: code/graphs/dfs-recursive.rs
fn dfs(g: &Graph, u: usize, seen: &mut Vec<bool>) {
    seen[u] = true;                 // visit u
    for &v in &g.adj[u] {           // each neighbor
        if !seen[v] {
            dfs(g, v, seen);        // recurse; backtrack on return
        }
    }
}
